### PR TITLE
Support `version` when generating an SDK

### PR DESCRIPTION
**Pull request by rcoh**
*Description of changes:*
This adds support for specifying `version` in `runtimeConfig` which will modify dependencies to the Runtime crates to use the specified version instead of a path dependency.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/CodegenVisitor.kt [hunk 1]
```diff
@@ -93,7 +93,10 @@ class CodegenVisitor(context: PluginContext, private val codegenDecorator: RustC
             "cargo fmt".runCommand(fileManifest.baseDir, timeout = 10)
         } catch (_: CommandFailed) {
             logger.warning("Generated output did not parse [${service.id}]")
+        } catch (ex: Exception) {
+            logger.warning("Other error trying to format: $ex")
```
Comment (jonhoo) [inline codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/CodegenVisitor.kt:97]: This looks like an unrelated change? May be suitable for a separate PR, or at least a call-out in this one?

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt [hunk 2]
```diff
@@ -9,27 +9,51 @@ import software.amazon.smithy.codegen.core.Symbol
...
+import software.amazon.smithy.rust.codegen.rustlang.Local
 import software.amazon.smithy.rust.codegen.rustlang.RustDependency
 import software.amazon.smithy.rust.codegen.rustlang.RustType
 import software.amazon.smithy.rust.codegen.rustlang.RustWriter
 import software.amazon.smithy.rust.codegen.rustlang.asType
 import java.util.Optional
 
-data class RuntimeConfig(val cratePrefix: String = "smithy", val relativePath: String = "../") {
+sealed class RuntimeCrateLocation {
+    data class Relative(val relativePath: String) : RuntimeCrateLocation()
```
Comment (jonhoo) [inline codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt:23]: Should this maybe be called `Path` instead? It feels like that's the operative description, as opposed to whether the path is relative or absolute.

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolMetadataProvider.kt [hunk 3]
```diff
@@ -93,8 +93,10 @@ class BaseSymbolMetadataProvider(base: RustSymbolProvider) : SymbolMetadataProvi
     }
 
     companion object {
-        private val defaultDerives = with(RuntimeType) {
-            listOf(Debug, PartialEq, Clone)
+        private val defaultDerives by lazy {
```
Comment (jonhoo) [inline codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolMetadataProvider.kt:96]: This also feels like a separate change?

**Review by jonhoo — APPROVED** [review]
This'll definitely come in handy going forward!

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/CodegenVisitor.kt (hunk 1 above)
Comment (rcoh) [reply codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/CodegenVisitor.kt:97]: oh yeah good catch. This fixes the hard crash when `cargo` can't be found

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt (hunk 2 above)
Comment (rcoh) [reply codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt:23]: yeah agreed, especially since the tests use an absolute path 😂

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolMetadataProvider.kt (hunk 3 above)
Comment (rcoh) [reply codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/SymbolMetadataProvider.kt:96]: this should fix that error message you got where it just generically told you it couldn't be initialized and didn't actually say it was because Rust runtime was missing
